File: Java/SleepyCowHerding.java
import java.io.*;
import java.util.*;

public class SleepyCowHerding {

    //static BufferedReader r = new BufferedReader(new InputStreamReader(System.in));
    //static PrintWriter pw = new PrintWriter(System.out);

    public static void main(String[] args) throws IOException {
        BufferedReader r = new BufferedReader(new FileReader("herding.in"));
        PrintWriter pw = new PrintWriter(new FileWriter("herding.out"));
		
        final int N = Integer.parseInt(r.readLine());

        int[] cows = new int[N];

        for (int i = 0; i < N; i ++) {
            cows[i] = Integer.parseInt(r.readLine());
        }
        Arrays.sort(cows);

        int minMoves = Integer.MAX_VALUE;

        if ((cows[1] - cows[0] > 2 && cows[N - 1] - cows[1] == N - 2) || (cows[N - 1] - cows[N - 2] > 2 && cows[N - 2] - cows[0] == N - 2)) {
            minMoves = 2;
        }
        else {
            int j = 0;
            for (int i = 0; i < N; i ++) {
                while (j + 1 < N && cows[j + 1] - cows[i] < N) {
                    j ++;
                }
                minMoves = Math.min(minMoves, N - (j - i + 1));
            }
        }

        int maxMoves = Math.max(cows[N - 2] - cows[0], cows[N - 1] - cows[1]) - (N - 2);

        pw.println(minMoves);
        pw.println(maxMoves);
        pw.close();
    }
}

File: Python/Sleepy_Cow_Herding.py
import sys

sys.stdin = open("herding.in","r")
sys.stdout = open("herding.out","w")


b,e,m = map(int,input().split())

sortedCows = [b,e,m]

sortedCows.sort()
minMoves = 0
maxMoves = 0

#minimum moves possible---------------------------------
if sortedCows[2] - sortedCows[1] == 1 and sortedCows[1] - sortedCows[0] == 1:
    print(0)
elif sortedCows[2] - sortedCows[1] == 2 or sortedCows[1] - sortedCows[0] == 2:
    print(1)
else:
    print(2)
    
print(max(sortedCows[2] - sortedCows[1], sortedCows[1] - sortedCows[0]) - 1)